packages/react-pose/src/components/Transition/children.ts

```diff
@@ -1,58 +1,69 @@
 import * as React from 'react';
 import { ReactElement } from 'react';
-import { CurrentPose } from '../PoseElement/types';
-import { MergeChildrenProps, Props } from './types';
+import { Props, State } from './types';
+import { invariant } from 'hey-listen';
 const { Children, cloneElement } = React;
 
-const getKey = (child: ReactElement<any>): string => child.key as string;
-
-const filterChildProps = ({ children, _pose, onPoseComplete, popFromFlow, ...props }: Props) => props;
+const filterChildProps = ({
+  children,
+  _pose,
+  onPoseComplete,
+  popFromFlow,
+  ...props
+}: Props) => props;
+
+const getKey = (child: ReactElement<any>): string => {
+  invariant(
+    child && child.key !== null,
+    'Every child of Transition must be given a unique key'
```

Comment:
is this true for a single child case?

Reply:
Yeah because if that single child switches out for another single child we need to know the difference.

Reply:
👌